uaclient/util.py

```diff
@@ -497,6 +497,54 @@ def configure_web_proxy(
         request.install_opener(opener)
 
 
+def configure_snap_proxy(
+    http_proxy: "Optional[str]", https_proxy: "Optional[str]"
+) -> None:
+    """
+    Configure snap to use http and https proxies.
+
+    :param http_proxy: http proxy to be used by snap. If None, it will
+                       not be configured
+    :param https_proxy: https proxy to be used by snap. If None, it will
+                        not be configured
+    """
+    if http_proxy:
+        subp(["snap", "set", "system", "proxy.http={}".format(http_proxy)])
+
+    if https_proxy:
+        subp(["snap", "set", "system", "proxy.https={}".format(https_proxy)])
+
+
+def configure_livepatch_proxy(
+    http_proxy: "Optional[str]", https_proxy: "Optional[str]"
+) -> None:
+    """
+    Configure livepatch to use http and https proxies.
+
+    :param http_proxy: http proxy to be used by livepatch. If None, it will
+                       not be configured
+    :param https_proxy: https proxy to be used by livepatch. If None, it will
+                        not be configured
+    """
+    if http_proxy:
```

Comment:
Same comment as on configure_snap_proxy above, this function needs to support unsetting http_proxy and or https_proxy in livepatch if either of those values are None or "".

Also,we need to perform the same pre-flight check for livepatch existing proxy settings before setting them.
Since we probably want a consolidated single prompt if livepatch and/or snap proxies are already set to something non-null it might actually make sense to consolidate both of these configure*proxy functions into one function that preflight checks livepatch and snap proxy settings and provides a single prompt  that accounts for any pre-existing proxy settings which will be overridden.